crates/bevy_diagnostic/src/diagnostic.rs

```diff
@@ -302,11 +298,22 @@ impl RegisterDiagnostic for App {
     ///     .add_plugins(DiagnosticsPlugin)
     ///     .run();
     /// ```
+    fn register_diagnostic(&mut self, diagnostic: Diagnostic) -> &mut Self;
```

Comment:
Nice catch here. I feel like there should be a clippy lint for documentation that is placed on a trait impl instead of a declaration.